### PR TITLE
verify token auth file

### LINKED ISSUE #55434: Blank tokens in  `--token-auth-file`
> /kind feature

**What happened**:
While following through this tls [bootstrapping guide](https://kubernetes.io/docs/admin/kubelet-tls-bootstrapping/#token-authentication-file) I accidentally made a file with blank tokens. (format was `,kubelet,kubelet`). I noticed that there was a particular format recommended in this guide but I don’t see that we do any verification of the token when [loading the file](https://github.com/kubernetes/apiserver/blob/master/pkg/authentication/token/tokenfile/tokenfile.go#L43). I also noted that you can enter any string as a token and authenticate.

**What you expected to happen**:
I'd expect the apiserver to verify the token length, or at least warn if no token is provided. I did verify that when the token is set to blank a user is not able to auth by providing an empty string as the token in `~.kube/config`.

**How to reproduce it (as minimally and precisely as possible)**:
Run the `kube-apiserver` with `--token-auth-file` pointing to a file with the value `,kubectl,kubectl`

**Anything else we need to know?**:
Nope.

**Environment**:
- Kubernetes version (use `kubectl version`):
```bash
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T23:15:59Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T22:55:19Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
```bash
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
verify token auth file and to prevent empty token.

https://kubernetes.io/docs/admin/kubelet-tls-bootstrapping/#token-authentication-file
https://kubernetes.io/docs/admin/authentication/#static-token-file

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #55434

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by bkochendorfer** [comment]
This fixes a blank token but should the token length also be verified? It's unclear to me if that is just a recommendation or if we should enforce the length.

**Comment by hzxuzhonghu** [comment]
@bkochendorfer I am not sure whether the token length is fixed.

**Comment by ncdc** [comment]
@kubernetes/sig-auth-pr-reviews 
/assign @liggitt @ericchiang @mattmoyer 
/unassign

**Comment by ericchiang** [comment]
Seems fine since we could hit some really weird edge cases accepting an empty bearer token.

I don't think it's worth the effort to do more validation.

/lgtm

File: staging/src/k8s.io/apiserver/pkg/authentication/token/tokenfile/tokenfile.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ func NewCSV(path string) (*TokenAuthenticator, error) {
 		if len(record) < 3 {
 			return nil, fmt.Errorf("token file '%s' must have at least 3 columns (token, user name, user uid), found %d", path, len(record))
 		}
+		if record[0] == "" {
+			return nil, fmt.Errorf("token file '%s' must have non empty token", path)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/authentication/token/tokenfile/tokenfile.go]: the auth layer won't actually authenticate an empty token. I'd probably warn and skip the record

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/authentication/token/tokenfile/tokenfile.go]: SGTM

**Comment by hzxuzhonghu** [comment]
@ericchiang @liggitt Done， warning if empty token and just ignore this line.

**Pull request merged**